src/lib/common/JsonHelper.cpp

```diff
@@ -422,17 +424,13 @@ void JsonVectorHelper::addNull(void)
 
 
 
+
 /* ****************************************************************************
 *
 * JsonVectorHelper::str -
 */
-std::string JsonVectorHelper::str()
+std::string JsonVectorHelper::str(void)
 {
-  // This check allows to call str() several times (needed when this is used in ExprContext)
-  if (!closed)
-  {
-    ss += ']';
-    closed = true;
-  }
+  ss += ']';
   return ss;
 }
```

Comment:
This was introduced in 4.0.0, but it seems is not actually needed.

Going back to 3.12.0 version of this code.

NTC (informative)